speech/detector.go

```diff
@@ -317,6 +220,11 @@ func (sd *Detector) Detect(pcm []float32) ([]Segment, error) {
 		}
 	}
 
+	// Close off the current segment
+	if sd.triggered && len(segments) > 0 {
+		segments[len(segments)-1].SpeechEndAt = float64(len(pcm)) / float64(sd.cfg.SampleRate)
+	}
```

Comment:
@cpoile The reason I left it undefined (0 really) is because the fact that we don't have more samples doesn't necessarily mean the speech has ended.

Reply:
@streamer45 I guess that's true outside of the data, but in terms of this audio the speech has ended (because the sample has ended), right? The transcriber special-cases a zero `SpeechEndAt` anyway... So I thought may as well just close it off here instead of there. But I can take it out if you'd rather we handle that in the transcriber side.

Reply:
My main worry is that if this library is ever used on a continuous stream this could be interpreted as a false positive. I suppose that we could simply document the method to explain this "quirk" and move on.

Reply:
Ok, I won't change it here since it's a library. I'll handle it in the transcriber.